vagrant/lib/forklift/box_loader.rb

```diff
@@ -35,5 +35,15 @@ def versions
       YAML.load_file("#{@root_dir}/config/versions.yaml")
     end
 
+    def box_config
+      config = if File.exist?("#{@root_dir}/config/boxes.yaml")
```

Comment:
just so that I understand this PR correctly:
the idea is to provide filters (based on os, foreman version and certain properties of the box) ontop of what is generated by building a matrix from `versions.yaml`, correct?
and then we provide multiple files that could be used as "profiles" (well, presets for those filters)?

I think I would find it easier if those filters be just exclude ones the user can set in their `settings.yaml`, like (this is what I'd probably set it to):
```
exclude_os:
  - centos7-fips
  - debian9
  - ubuntu1604
  - ubuntu1804

exclude_foreman:
  - '1.20'
  - '1.21'
```

Maybe mixed with the idea from @ekohl to have certain configurations marked as "unsupported" and thus not displayed either unless the user sets `show_unsupported` to true -- this would eliminate the need for `exclude_foreman` *for me*.